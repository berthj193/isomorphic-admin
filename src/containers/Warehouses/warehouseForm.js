import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Form, { FormSubmit, FormDivider } from '../../components/uielements/form';
import OptionsInput from '../../components/uielements/optionsInput';
import Input from '../../components/uielements/input';
import { required, fieldListRequired } from '../../helpers/formValidation';
import withReduxForm from '../../components/formWrapper/withReduxForm';

export class WarehouseForm extends React.Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
  };

  render() {
    return (
      <Form onSubmit={this.props.handleSubmit}>
        <Field label="Name" name="name" component={Input} validate={[required]} required />
        <Field label="Export class" name="export_class" component={Input} validate={[required]} required />
        <FormDivider />
        <Field
          name="options"
          component={OptionsInput}
          required
          validate={[fieldListRequired]}
        />
        <FormDivider />
        <FormSubmit />
      </Form>
    );
  }
}

const form = withReduxForm({
  form: 'warehouseForm',
  enableReinitialize: true,
})(WarehouseForm);

export default form;
