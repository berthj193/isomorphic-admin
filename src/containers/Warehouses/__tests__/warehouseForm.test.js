import React from 'react';
import { WarehouseForm } from '../warehouseForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<WarehouseForm />', () => {
  const getComponent = (props = {}) => shallow(<WarehouseForm {...props} />);

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });
});
