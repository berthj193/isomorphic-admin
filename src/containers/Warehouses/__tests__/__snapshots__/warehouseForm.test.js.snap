// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarehouseForm /> should render properly 1`] = `
<Form
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    label="Name"
    name="name"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Export class"
    name="export_class"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <FormDivider />
  <Field
    component={[Function]}
    name="options"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <FormDivider />
  <FormSubmit />
</Form>
`;
